proposals/044-integration-model.md

```diff
@@ -0,0 +1,203 @@
+
+## Integration Model
+
+* Issue: https://github.com/syndesisio/syndesis-project/issues/22
+* Sprint: 14
+* Affected Repos:
+  - syndesis-rest
+  - syndesis-ui
+
+## Objective
+
+Currently integrations that are deployed are completely disconnected from the Syndesis DB afterwards. There is no way yet to update a running integration nor to see the state of the currently running integration.
+
+The purpose of this document is to suggest a concise state model for `Integrations` as well as define the interaction between the `Integration`s designed in _syndesis-ui_ with the integrations running in the OpenShift cluster whose creation is triggered by a deployment action by the user.
+
+When rethinking the integration and the state model, the following points should be considered:
+
+* Multiple versions of an Integration must be able to exist.
+* Exactly one of these integrations can be active.
+* Each deployment of an Integration creates a new version.
+* Users can switch integration versions from the UI
+* Versioned Integrations can be inspected from a history page.
+* Old integrations can't be changed but can be used as a blueprint for new integrations.
+* As soon as an integration is changed, a new version of an integration is created. All changes go to this new integration until it is deployed by the user. After this, its gets a fixed version and cannot be changed anymore.
+* The user must always be able to see and examine the currently active integration along with its version number.
+
+_Attention: THIS IS WORK-IN-PROGRESS. The propsoal is PRed early to collect early feedback_
+
+
+## Integration state model
+
+Technical an Integration is the domain object describing a Camel Route. 
+It is managed by `syndesis-rest` and `syndesis-ui`. 
+During design time this object only lives in this domain. 
+After a "Publish" operation the integration is transformed in an runnable artefact. 
+This runnable artefact consists of a GitHub repo containing the code and configuration and an OpenShift resource configuration which transforms this code into a running application, backed by an integration pod.
+Running integrations can be suspended and resumed without destroying the deployment.
+
+
+![Integration state model](images/integration-state.png "Integration State Model")
+
+| State       | Description |
+| ----------- |------------ |
+| Draft | Initial state of an integration. The integration is not yet deployed |
+| Active | Integration is deployed and running |
+| Inactive | Integration is deployed but is not running|
+| Undeployed | An integration has been undeployed | 
+| Error | The integration is deployed but in an error state |
+
+### Draft
+
+Right after an integration is created it enters the _Draft_ state. This means 
+the integration is not yet deployed. An integration leaves the draft state either by being published or deleted (in which case the whole integration is removed)
+
+After an activation there is no way back to the _Draft_ state.
```

Comment:
E.g. in _Draft_ the integration must not validate it simply a state for 'work-in-progress'. The most important motivation for not allowing it to go back to _Draft_ once out of the state is for a simplified operational mode as the integration then becomes immutable afterwards. When changing such an integration, a new Integration object with  the same name, but another version will be created, again in state _Draft_.

Yes, unfinished integrations are in state _Draft_.